components/org.wso2.carbon.identity.conditional.auth.functions.http/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/http/HTTPPostFunctionImpl.java

```diff
@@ -87,6 +87,24 @@ public void httpPost(String endpointURL, Object... params) {
                             "(Map<String, Object>) respectively.");
                 }
                 break;
+            case 4:
+                if (params[0] instanceof Map && params[1] instanceof Map && params[2] instanceof Map && params[3] instanceof Map) {
+                    Map<String, Object> map = (Map<String, Object>) params[0];
+                    if (map.get("type") == null || map.get("properties") == null) {
+                        throw new IllegalArgumentException("Invalid argument type. Expected AuthConfigModel.");
+                    }
+                    String type = (String) map.get("type");
+                    Map<String, Object> propertiesMap = (Map<String, Object>) map.get("properties");
+                    authConfig = new AuthConfigModel(type, propertiesMap);
+                    payloadData = (Map<String, Object>) params[1];
+                    headers = validateHeaders((Map<String, ?>) params[2]);
+                    eventHandlers = (Map<String, Object>) params[3];
+                }  else {
+                    throw new IllegalArgumentException("Invalid argument type. Expected payloadData " +
+                            "(Map<String, Object>), headers (Map<String, String>), and eventHandlers " +
+                            "(Map<String, Object>) respectively.");
+                }
+                break;
```

Comment:
I see the same ordering problem I mentioned above.

Also, better have a function that can process the authconfig

Reply:
Fixed with https://github.com/wso2-extensions/identity-conditional-auth-functions/pull/157/commits/dc7ea3e7070ae1a51e26d6d777bb50b4e410581b